src/Confluent.Kafka/Impl/Metadata.cs

```diff
@@ -90,4 +90,61 @@ struct rd_kafka_group_list
         internal IntPtr groups;
         internal int group_cnt;
     };
+    
+    enum rd_kafka_vtype
+    {
+        End,       // va-arg sentinel
+        Topic,     // (const char *) Topic name
+        Rkt,       // (rd_kafka_topic_t *) Topic handle
+        Partition, // (int32_t) Partition
+        Value,     // (void *, size_t) Message value (payload)
+        Key,       // (void *, size_t) Message key
+        Opaque,    // (void *) Application opaque
+        MsgFlags,  // (int) RD_KAFKA_MSG_F_.. flags
+        Timestamp, // (int64_t) Milliseconds since epoch UTC
+        Header,    // (const char *, const void *, ssize_t) Message Header
+        Headers,   // (rd_kafka_headers_t *) Headers list
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    struct rd_kafka_key_val
+    {
+        public IntPtr data;
+        public UIntPtr size;
+    }
+    
+    [StructLayout(LayoutKind.Explicit, Size = 64)]
+    struct rd_kafka_vu_data
+    {
+        [FieldOffset(0)]
+        public IntPtr topic;
+
+        [FieldOffset(0)]
+        public int partition;
+
+        [FieldOffset(0)]
+        public rd_kafka_key_val key;
```

Comment:
you don't actually use this above (you use `val`), suggest getting rid of this and calling it `mem` to match the librdkafka union.

Reply:
Not sure I'm following.. ;(